test/unit/tasks/usage_limits_test.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class UsageLimitsTest < ActiveSupport::TestCase
+  include ActiveJob::TestHelper
+
+  test 'executing the task removes orphaned usage limits' do
+    service = FactoryBot.create(:simple_service)
+    service_metric = service.metrics.first
+    plan = FactoryBot.create(:application_plan_without_rules, issuer: service)
+
+    backend_api1 = FactoryBot.create(:backend_api, account: service.account)
+    backend_metric1 = FactoryBot.create(:metric, owner: backend_api1)
+    backend_usage_limit1 = FactoryBot.create(:usage_limit, metric: backend_metric1, plan: plan)
+
+    backend_api2 = FactoryBot.create(:backend_api, account: service.account)
+    backend_metric2 = FactoryBot.create(:metric, owner: backend_api2)
+    backend_usage_limit2 = FactoryBot.create(:usage_limit, metric: backend_metric2, plan: plan)
+    FactoryBot.create(:backend_api_config, service: service, backend_api: backend_api2)
+
+    service_usage_limit = FactoryBot.create(:usage_limit, metric: service_metric, plan: plan)
+
+    assert_same_elements [backend_usage_limit1, backend_usage_limit2, service_usage_limit], plan.usage_limits
+
+    execute_rake_task 'usage_limits.rake', 'usage_limits:clean_orphans'
+
+    # usage limit that belongs to backend 1 should go away, because backend 1 is not linked to the service
+    # other limits should stay, because they are valid
+    assert_same_elements [backend_usage_limit2, service_usage_limit], plan.reload.usage_limits
```

Comment:
[Sorry, I wrote this comment to the commit and not the PR. Moving it to here...]

I don't understand this assertion. If `backend_usage_limit1` is deleted, we should assert that `backend_usage_limit2` and `service_usage_limit` are still in DB but `backend_usage_limit1` has been deleted from DB.

With this assertion, `backend_usage_limit2` and `service_usage_limit` could be the same but not be equal to before running the rake task. And we are not checking what happened with `backend_usage_limit1`.

Or am I missing something?

Reply:
OK my mistake. I interpreted this as "assert `backend_usage_limit2` is equal to `service_usage_limit`". It's fine 👍